modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -93,7 +93,7 @@ func (k Keeper) OnChanOpenAck(
 	}
 
 	k.SetActiveChannelID(ctx, portID, channelID)
-	k.SetInterchainAccountAddress(ctx, portID, metadata.Address)
+	k.SetInterchainAccountAddress(ctx, metadata.HostConnectionId, portID, metadata.Address)
```

Comment:
controller docs should specify this is the host connection ID (currently says "associated" connection)

Reply:
Agreed. I'll update the godoc to be more explicit when its decided whether to use the controller or host connection ID here when setting state.

Reply:
Updated to use controller connection ID here as agreed